library/alloc/src/boxed/thin.rs

```diff
@@ -171,7 +171,7 @@ struct WithHeader<H>(NonNull<u8>, PhantomData<H>);
 /// An opaque representation of `WithHeader<H>` to avoid the
 /// projection invariance of `<T as Pointee>::Metadata`.
 #[repr(transparent)]
-struct WithOpaqueHeader(NonNull<u8>);
+struct WithOpaqueHeader(#[allow(unused_tuple_struct_fields)] NonNull<u8>);
```

Comment:
For future readers of this code, it's probably helpful to add a comment explaining why this field is 'unused'.

```suggestion
#[allow(unused_tuple_struct_fields)] // Field only used through `WithHeader` type above.
struct WithOpaqueHeader(NonNull<u8>);
```

Reply:
Thanks, tweaked!